lib/web/apiserver.go

```diff
@@ -867,8 +869,11 @@ func (h *Handler) bindDefaultEndpoints() {
 	h.POST("/webapi/tokens", h.WithAuth(h.upsertTokenHandle))
 	// used for updating a token
 	h.PUT("/webapi/tokens", h.WithAuth(h.upsertTokenHandle))
+	// TODO(kimlisa): DELETE IN 19.0 - Replaced by /v2/webapi/token endpoint
 	// used for creating tokens used during guided discover flows
 	h.POST("/webapi/token", h.WithAuth(h.createTokenForDiscoveryHandle))
+	// used for creating tokens used during guided discover flows
+	h.POST("/v2/webapi/token", h.WithAuth(h.createTokenForDiscoveryHandle))
```

Comment:
Not sure this will align well with our weird v1 auto prefix:
https://github.com/gravitational/teleport/blob/c222030e3f2eff643d90d30f4b634482fa896ba2/lib/web/apiserver.go#L618-L622

This /v2/ will make `/v1/v2/webapi/token` to work as well 😆 

We need to fix the weird prefixing issue first

Reply:
One possible way is to inverse the handler.

Append the `/v1/` to every single route `h.*("/v1..."`.

```go
 if strings.HasPrefix(r.URL.Path,"/webapi") { 
 	addPrefix(apiPrefix, h).ServeHTTP(w, r) 
 	return 
 } 


func addPrefix(prefix string, h Handler) Handler {
	if prefix == "" {
		return h
	}
	return http.HandlerFunc(func(w ResponseWriter, r *Request) {
		p := path.Join(prefix,r.URL.Path)
		rp := path.Join(prefix,r.URL.RawPath)
	
			r2 := new(Request)
			*r2 = *r
			r2.URL = new(url.URL)
			*r2.URL = *r.URL
			r2.URL.Path = p
			r2.URL.RawPath = rp
			h.ServeHTTP(w, r2)
		
	})
}
```

Reply:
thanks for catching that, i completely overlooked this part

looks like @avatus already thought of that though in his [RFD](https://github.com/gravitational/teleport/blob/master/rfd/0179-api-versioning.md#incrementing-version-prefix).

I'll apply his logic and test things out